src/container/__tests__/RelayContainer-test.js

```diff
@@ -310,18 +325,42 @@ describe('RelayContainer', function() {
   it('creates resolvers for each query prop with a fragment pointer', () => {
     RelayTestRenderer.render(
       () => <MockContainer foo={mockFooPointer} />,
+      mockRelayContext,
       mockRoute
     );
     expect(GraphQLStoreQueryResolver.mock.instances.length).toBe(1);
 
     RelayTestRenderer.render(
       () => <MockContainer foo={mockFooPointer} bar={[mockBarPointer]} />,
+      mockRelayContext,
       mockRoute
     );
     // `foo` resolver is re-used, `bar` is added
     expect(GraphQLStoreQueryResolver.mock.instances.length).toBe(2);
   });
 
+  it('recreates resolvers when relay context changes', () => {
+    var mockRelayContextA = new RelayContext();
+    var mockRelayContextB = new RelayContext();
+
+    RelayTestRenderer.render(
+      () => <MockContainer foo={mockFooPointer} />,
+      mockRelayContextA,
+      mockRoute
+    );
+    RelayTestRenderer.render(
+      () => <MockContainer foo={mockFooPointer} />,
+      mockRelayContextB,
+      mockRoute
+    );
+
+    const mockResolvers = GraphQLStoreQueryResolver.mock.instances;
+    expect(mockResolvers.length).toBe(2);
+    expect(mockResolvers[1].mock.store).toBe(mockRelayContextB.getStoreData());
+    expect(mockResolvers[0].reset).toBeCalled();
```

Comment:
These are great tests. My only feedback is that nothing is tested between the first and second render call - ideally we'd verify the state after the first render and then test how it changed after the second render.